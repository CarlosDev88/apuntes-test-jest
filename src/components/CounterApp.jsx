import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value = 10, factor = 1 }) => {
  const [counter, setcounter] = useState(value);

  const handleAdd = () => {
    setcounter(counter + factor);
  };

  const handleRemove = () => {
    setcounter(counter - factor);
  };

  const reset = () => {
    setcounter(value);
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+{factor}</button>
      <button onClick={reset}>Reset</button>
      <button onClick={handleRemove}>-{factor}</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
  factor: PropTypes.number,
};

CounterApp.defaultProps = {
  factor: 1,
};
