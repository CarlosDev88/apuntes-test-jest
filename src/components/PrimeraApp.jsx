import React from "react";

import PropTypes from "prop-types";

export const PrimeraApp = ({ saludo, subtitulo }) => {
  //   const { saludo } = props;
  let arr = [1, 2, 3, 4, 5, 6];
  let objeto = {
    nobre: "pepe",
    edad: 25,
  };
  return (
    <>
      <div>
        <h1>Esto es el primer functional component</h1>
        <h2>{arr.map((num) => `${num},`)}</h2>
        {arr.map((num) => ` ${num * 2},`)}
        <pre>{JSON.stringify(objeto)}</pre>
      </div>

      <div>
        <h2> Otra parte del componente </h2>
        {saludo}
        <p>{subtitulo}</p>
      </div>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
  subtitulo: PropTypes.string,
};

PrimeraApp.defaultProps = {
  subtitulo: "soy un subtitulo",
};
