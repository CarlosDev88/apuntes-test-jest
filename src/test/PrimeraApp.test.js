import React from "react";
import "@testing-library/jest-dom";
import { PrimeraApp } from "../components/PrimeraApp";
import { shallow } from "enzyme";

describe("pruebas componente <PrimeraApp/>", () => {
  //   test("deve mostrar 'Hola soy pepe'", () => {
  //     const saludo = "Hola soy pepe";
  //     const warapper = render(<PrimeraApp saludo={saludo} />);

  //     expect(warapper.getByText(saludo)).toBeInTheDocument();
  //   });

  test("deve mostrar <PrimeraApp/> correctamente", () => {
    const saludo = "Hola soy pepe";
    const warapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(warapper).toMatchSnapshot();
  });

  test("deve mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola soy pepe";
    const subtitulo = "Amigo que pasa";
    const warapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textoParrafo = warapper.find("p").text();
    expect(textoParrafo).toEqual(subtitulo);
  });
});
