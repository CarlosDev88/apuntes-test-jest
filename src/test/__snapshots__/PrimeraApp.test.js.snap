// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas componente <PrimeraApp/> deve mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <div>
    <h1>
      Esto es el primer functional component
    </h1>
    <h2>
      1,
      2,
      3,
      4,
      5,
      6,
    </h2>
     2,
     4,
     6,
     8,
     10,
     12,
    <pre>
      {"nobre":"pepe","edad":25}
    </pre>
  </div>
  <div>
    <h2>
       Otra parte del componente 
    </h2>
    Hola soy pepe
    <p>
      soy un subtitulo
    </p>
  </div>
</Fragment>
`;
